site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,666 @@
+                        cy.wrap(row).children().eq(2).then((user_id) => {
+                            const id = user_id.text().trim();
+                            all_late_users[g_id][id] = null;
+                            all_user_ids.push(id);
+                        });
+                    });
+            });
+
+            it('should have days submitted late > 0', () => {
+                for (const user_id in all_late_users[g_id]) {
+                    cy.visit([]);
+                    cy.login(user_id);
+                    cy.visit(['sample', 'late_table']);
+
+                    // Wait for login change to take place
+
+                    // Find # of days late within the row in the late day usage table
+                    cy.get('td[data-before-content="Event/Assignment"]')
+                        .contains(g_title)
+                        .siblings('td[data-before-content="Days Submitted Late"]')
+                        .should('not.have.value', 'N/A').then((value) => {
+                            // Grab the # of days late (should be > 0)
+                            const days_late = parseInt(value.text());
+                            expect(days_late).to.be.greaterThan(0);
+                            all_late_users[g_id][user_id] = days_late;
+
+                            // Wait for login change to take place
+                            cy.wait(3000);
+                            // Generate predicted status
+                            const status = predictedStatus(days_allowed, days_late, 0);
+
+                            // Find late day status within the row in the late day usage table
+                            cy.get('td[data-before-content="Event/Assignment"]')
+                                .contains(g_title)
+                                .siblings('td[data-before-content="Status"]')
+                                .contains(status)
+                                .should('exist');
+                        });
+                    cy.logout();
+                }
+            });
+
+            it('should have 0 late days used on bulk late days table', () => {
+                cy.visit(['sample', 'bulk_late_days']);
+                cy.login('instructor');
+
+                // All statuses should be bad, 0 late days should be charged
+                for (const user_id in all_late_users[g_id]) {
+                    cy.get(`[data-user="${user_id}"] > [data-before-content="${g_title}"]`)
+                        .contains('0')
+                        .should('exist');
+                }
+            });
+        });
+    }
+
+    describe('Test changes to late days allowed table', () => {
+        const late_days_remaining = {};
+        it('should grant students with 2 late days', () => {
+            cy.visit(['sample', 'late_days']);
+            cy.login('instructor');
+
+            for (const user_id of all_user_ids) {
+                const days = Math.floor(Math.random() * 2) + 1;
```

Comment:
Random numbers (even small ones) are considered unsafe, as one number can fail, while the others pass, making the test seem flaky.